### PR TITLE
APR-2913 Load people in control in to enhanced data structure

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 1]
```diff
@@ -46,12 +55,122 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
...
+                    {
+                        var orgPersonnel = new OrganisationPersonnel
+                        {
+                            OrganisationId = organisationId,
+                            PersonnelType = (int)personnelType,
+                            Name = submittedAnswersOrganisationNameSoleTrade.FirstOrDefault().Answer
+                        };
+                        foreach (SubmittedApplicationAnswer record in person)
+                        {
+                            var dobArray = record.Answer.Split(",");
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:126]: I think you need a bit of protective code to check this isn't null, and it contains 2 and 2 only values in the array

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 2]
```diff
@@ -46,12 +55,122 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
...
+                        PersonnelType = (int)personnelType,
+                    };
+                    foreach (var record in person)
+                    {
+                        switch (record.ColumnHeading)
+                        {
+                            case "Name":
+                                orgPersonnel.Name = record.Answer;
+                                break;
+                            case "Date of birth":
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:160]: Have you smoke tested this? |I thought the data was of the form "Month Year", and not sure about the conversion working.  If you've smoke tested and all ok, just resolve this conversation

**Review by MarkFCain — APPROVED** [review]
some minor comments and a trivial request to maybe add some protective code. Approved.

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 3]
```diff
@@ -50,16 +56,127 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
...
+                            Name = submittedAnswersOrganisationNameSoleTrade.FirstOrDefault().Answer
+                        };
+                        foreach (SubmittedApplicationAnswer record in person)
+                        {
+                            if(record.Answer != null)
+                            {
+                                var dobArray = record.Answer.Split(",");
+                                if(dobArray.Length == 2)
+                                {
+                                    orgPersonnel.DateOfBirthMonth = byte.Parse(dobArray[0]);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:131]: Why byte parse?

Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: This is changed to int already

File: src/SFA.DAS.Roatp.Functions/ApplyTypes/OrganisationPersonnel.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace SFA.DAS.Roatp.Functions.ApplyTypes
+{
+    public class OrganisationPersonnel
+    {
+        public int Id { get; set; }
+        public Guid OrganisationId { get; set; }
+        public int PersonnelType { get; set; }
```
[the remark this answers is not in the document]
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplyTypes/OrganisationPersonnel.cs]: Updated

File: src/SFA.DAS.Roatp.Functions/Infrastructure/Databases/ApplyDataContext.cs [hunk 5]
```diff
@@ -95,6 +95,11 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .HasForeignKey(cf => cf.ApplicationId)
                     .OnDelete(DeleteBehavior.NoAction);
             });
+
+            modelBuilder.Entity<OrganisationPersonnel>(entity =>
+            {
+                entity.ToTable("OrganisationPersonnel");
```
[the remark this answers is not in the document]
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/Infrastructure/Databases/ApplyDataContext.cs]: Updated the code

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 6]
```diff
@@ -50,16 +58,126 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
...
+                _logger.LogInformation($"Organisation Personnel successfully load for application {applicationId}");
+            }
+
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Unable to load Organisation Personnel for Application: {applicationId}");
+            }
+        }
+
+        private static List<OrganisationPersonnel> ExtractTabularAnswerOrganisationPersonnel(List<SubmittedApplicationAnswer> answers, Guid organisationId, string questionId, PersonnelType personnelType)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:108]: Looks so much better now 💯

**Review by DevenKShah — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**